starknet_py/net/client_models.py

```diff
@@ -0,0 +1,139 @@
+from dataclasses import dataclass
+from enum import Enum, auto
+from typing import List, Any, Dict, Optional
+
+from starkware.starknet.services.api.gateway.transaction import (
+    InvokeFunction as IF,
+    Transaction as T,
+    ContractDefinition as CD,
+    Deploy as D,
+)
+
+
+from starknet_py.utils.docs import as_our_module
+
+InvokeFunction = as_our_module(IF)
+StarknetTransaction = as_our_module(T)
+ContractDefinition = as_our_module(CD)
+Deploy = as_our_module(D)
+
+
+@dataclass
+class Event:
+    from_address: int
+    keys: List[int]
+    data: List[int]
+
+
+@dataclass
+class L1toL2Message:
+    l1_address: int
+    l2_address: int
+    payload: List[int]
+
+
+@dataclass
+class L2toL1Message:
+    l2_address: int
+    l1_address: int
+    payload: List[int]
+
+
+class TransactionType(Enum):
+    INVOKE = 0
+    DEPLOY = auto()
+
+
+@dataclass
+class Transaction:
+    """
+    Dataclass representing a transaction
+    """
+
+    hash: int
+    contract_address: int
+    calldata: List[int]
+    entry_point_selector: int
+    transaction_type: TransactionType
+    version: int = 0
+    max_fee: int = 0
+
+
+class TransactionStatus(Enum):
+    UNKNOWN = 0
+    RECEIVED = auto()
+    PENDING = auto()
+    ACCEPTED_ON_L2 = auto()
+    ACCEPTED_ON_L1 = auto()
+    REJECTED = auto()
+
+
+@dataclass
+class TransactionReceipt:
+    """
+    Dataclass representing details of sent transaction
+    """
+
+    hash: int
+    status: TransactionStatus
+    events: List[Event]
+    l2_to_l1_messages: List[L2toL1Message]
+    l1_to_l2_consumed_message: Optional[L1toL2Message] = None
+    block_number: Optional[int] = None
```

Comment:
This is how it's done in cairo-lang, I guess we can do that. There will be a problem with full node client, as realistically it will only be able to return base `Transaction` still.

Reply:
Don't you think it will eventually create a complicate structure of dataclasses? 

Because realistically we should have receipt for accepted transaction, pending, rejected etc.

Reply:
Ok, let's leave it like this for now, but ideally we should properly model the result.